src/raftstore/store/peer_storage.rs

```diff
@@ -429,17 +429,28 @@ impl PeerStorage {
     // Delete all data belong to the region. Results are stored in `wb`.
     pub fn clear(&self, wb: &mut WriteBatch) -> Result<()> {
         let timer = Instant::now();
-        for r in self.region_key_ranges() {
-            try!(self.engine.scan(&r.0,
-                                  &r.1,
-                                  &mut |key, _| {
-                                      try!(wb.delete(key));
-                                      Ok(true)
-                                  }));
+
+        for (start_key, end_key) in self.region_key_ranges() {
+            // meta range only has little keys, no need use unsafe delete.
+            if !start_key.starts_with(keys::REGION_META_PREFIX_KEY) {
+                // TODO:
+                // 1. need to handle `delete_in_range` successfully but following
+                // WriteBatch failed.
+                // 2. need to care other CF?
```

Comment:
If `delete_in_range` can finish fast, then it doesn't matter whether the data is small or not. It doesn't delete extra data and keep the code simple.